### PR TITLE
SQ-288/Select page based on the current date

### PR DESCRIPTION
**Pull request by emartynov**
Sorry, for this is blind PR:

1. I can not build app unless I fake all secrets or create my own
2. I can not add test since there are no tests for the UI parts

I might add Robolectric or Espresso test, but both have a drawback. Robolectric has not respected library in the community. For the espresso, it is not enough just to add it to be able to run locally but it should be also CI autorun, that requires starting the emulator.

Let me know what do you think. Feel free to just ignore this PR.

And sleep well since it is 2 AM in London.

### CODE REVIEW

**Comment by rock3r** [comment]
Thanks @emartynov for the PR! We'll look at it as soon as possible.

File: app/src/main/java/net/squanchy/schedule/SchedulePageView.java [hunk 1]
```diff
@@ -59,12 +69,28 @@ protected void onFinishInflate() {
...
         viewPager.setAdapter(viewPagerAdapter);
+        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+                //nothing to do
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                currentPage = position;
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/SchedulePageView.java:86]: Isn't this triggered when calling `viewPager.setCurrentItem()`? I never remember lol

File: app/src/main/java/net/squanchy/schedule/SchedulePageView.java [hunk 2]
```diff
@@ -59,12 +69,28 @@ protected void onFinishInflate() {
...
         progressBar = findViewById(R.id.progressbar);
 
-        ViewPager viewPager = findViewById(R.id.viewpager);
+        viewPager = findViewById(R.id.viewpager);
         TabLayout tabLayout = findViewById(R.id.tabstrip);
         tabLayout.setupWithViewPager(viewPager);
         hackToApplyTypefaces(tabLayout);
 
         viewPager.setAdapter(viewPagerAdapter);
+        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/SchedulePageView.java:78]: This could use some extracting

**Review by rock3r — COMMENTED** [review]
Looks ok, just a couple of comments — one of which is to myself to check out how it actually works. Thanks again for the PR, will test it out asap

**Comment by rock3r** [comment]
BTW this is part of the story of #288

File: app/src/main/java/net/squanchy/schedule/SchedulePageView.java (hunk 1 above)
Comment (emartynov) [reply app/src/main/java/net/squanchy/schedule/SchedulePageView.java:86]: It might! It was late night and I didn't bother to manual test nor write test.

**Comment by emartynov** [comment]
I would vote to trash this PR and wait for the story to be done! Droidcon is out so this issue will not annoy us till next year. Thanks for having me on air!

**Pull request closed without merging**